src/main.rs

```diff
@@ -244,7 +282,41 @@ impl eframe::App for App {
                     Route::Equipment => equipment(ui, &mut self.vm),
                     Route::Inventory => inventory(ui, &mut self.vm),
                     Route::EventFlags => events(ui, &mut self.vm),
-                    Route::Regions => regions(ui, &mut self.vm),
+                    Route::Regions => {
```

Comment:
This should be moved to the ui/regions as part of the  `regions(ui, &mut self.vm)`. Currently this splits up the ui code. Some of it is here then the rest of it is called from `regions(ui, &mut self.vm)`